index.js

```diff
@@ -9,10 +9,20 @@ from2.obj = obj
 var Proto = ctor()
 
 function from2(opts, read) {
-  if (typeof opts === 'function') {
+  if (typeof opts !== 'object' || Array.isArray(opts)) {
     read = opts
     opts = {}
   }
+  
+  if(Array.isArray(read)) {
+    read = read.slice()
+    return from2(function () {
+      if(read.length)
+        this.push(read.shift())
+      else
+        this.push(null)
```

Comment:
use the callback provided by from2 instead of using push

``` js
from2(function(size, cb) {
  cb(null, read.length ? read.shift() : null)
})
```